### ISSUE TITLE
BIP84 Single address wallet: Import format or derivation path ist wrong

### ISSUE THREAD

**Issue by Abstraktikus** [BlueWallet/BlueWallet#1440]
1.)	Go to https://iancoleman.io/bip39/
2.)	Create a 24 Word seed
“elephant minor kick boost suit heart review sponsor hospital square arm together submit transfer inch warrior pitch advice spray source solve cause observe cover”
3.)	Choose derivation path BIP84
4.)	Scroll down to derived addresses
5.)	Check the first address: "bc1qe8agy2jhmjmnrx6awxa5tsrrt5mtsjcdmpj7xk"
6.)	Make a mouse over the associated private key: "L32SxAktcrv9sWqhaN2R7BJBWaCQhr1ahhDPAkCvfmmkLjKQfMYL"
7.)	The private-key barcode appears. Import this Barcode in BlueWallet:

![L32SxAktcrv9sWqhaN2R7BJBWaCQhr1ahhDPAkCvfmmkLjKQfMYL](https://user-images.githubusercontent.com/17813711/89340647-59ec6f00-d6a0-11ea-9131-611e63d19573.png)

8.)	Press “Receive"
9.)	The showed imported address is “389c1VeCPaDFRKfeJkG2tDUhKPTf8XvHt7” instead of “bc1qe8agy2jhmjmnrx6awxa5tsrrt5mtsjcdmpj7xk”

--> Import Format (or derivation path) is wrong!

**Comment by ncoelho** [comment]
The wallet needs to have transactions. Otherwise all combinations are
possible and all wallets are valid

**Comment by Abstraktikus** [comment]
This feels unbelievable strange because the private keys AND the barcodes looks different for each BIP-Format. In this case the UI should at least ask for the wished derivation path out of supported variants e.g. by showing the address variants for the importet priv-key:  
•	1PD6…Hsrj (BIP32)
•	13Lv….uUt (BIP44)
•	3CmU….K8Lf (BIP49)
•	bc1q….yx27 (BIP84)
OR the wallet should support all TRX-Variants and use the newest format for sending and receiving. 
OR (easier but can be wrong) the wallet should search for transactions in all BIP-Formats and if no TRX can be found, use the newest format.

![Barcodes](https://user-images.githubusercontent.com/17813711/89385567-1aee0600-d700-11ea-90ef-d20e74444198.png)

**Comment by Abstraktikus** [comment]
Supplement: Or different behaviors may vary depending on whether the "advanced mode" is selected.

**Comment by Overtorment** [comment]
WIFs are an old format and dont carry a hint what kind of address should be unwrapped from the private key, so BW makes intelligent guess by trying our all possible addresses and whichever has balance - will be imported. If you put some satoshis  on that bc1 address and retry importing WIF it will be imported as bech32 single-address wallet okay.
If guessing fails the default one is _wrapped  segwit_ which starts with `3`, which is a legacy behaviour (because a long time ago, BW could work only with single wrapped segwit addresses).
There's no BIP84 or HD wallet stuff in general involved in this whole process.

In short, this is not a bug. Could put this stuff in FAQ tho @ncoelho

**Comment by Abstraktikus** [comment]
Thanks for the explanation. Interessting. Hmm... even if there is a good explanation for this. Incomprehensible behaviour when dealing with "wallets" weakens confidence in the app. Hence, if no money is found, for my taste, I would suggest to reject the import with an business-error: 

"The used barcode does not contain a version information. To determine the format to use, a pre-deposit is required.
1.) Use an existing wallet to pay a few satoshis to the single address 
2.) Wait for a confirmation
3.) Re-import the associated private key."

The motive for this suggestion is principal: A good process provides its guidance as a component, not in the FAQ.
Please excuse my tenacity. If BlueWallet weren't the best and therefore my most loved bitcoin wallet, this little thing wouldn't be worth a line ;-)

**Comment by ncoelho** [comment]
I like the idea, what about something more simple, like just a message explaining what happen:

"We haven't identified any transaction on the imported wallet, so a default wallet type was created for you, make sure to backup your new mnemonic seed"

**Comment by Abstraktikus** [comment]
In principle, I like the idea of a simple text. But to find a short, comprehensible language for this complex is not easy.
1.) There is no need for a backup, as a single address wallet gets imported from an external source.
2.) The source of the private keys is totally unclear. If someone imported a paper wallet, it is probably in most cases fine to just import it. But if someone imported a private key out of a HD-wallet, the format should match the BIP-Standard and derivation path for this mnemonic seed. There are again multiple reasons why a person would do this. (E.g. to separate the purpose or sources of payments within a wallet, or someone use only one address for outgoing payments out of a watch only wallet, or a shopkeeper uses an address for each cashier to distribute the stock in the morning and to withdraw sales in the evening.)

The answer must be non-technical but allow to recognize the problem and its solution. E.g.: “The selected address is [bc1q….yx27 (BIP84)]. If the chosen address is incorrect, deposit to the correct address first, then re-import the private key again.”
We haven't…. sounds centralized… ;-) As a non-native English speaker, I should not try to make a proposal for this sentence anyway.

Just for fun outlook: In a future where BlueWallet offers an annual subscription for its small business solution, featuring a more advanced address management per cashier for shopkeeper of thrift shops, market stalls, coiffeurs etc. which allows to open a withdrawal-limited lightning-wallets for each cashers with an auto- withdraw sales after closing time, there will be no longer incentives to operate with bitcoin single addresses. Until this is delivered, from the point of view of shopkeepers, traceability (at the product, category, or seller level) is more important than the privacy of their customers.

**Comment by Overtorment** [comment]
closing as nothing to do in this ticket

**Comment by hugoender** [comment]
I tried importing a BIP-84 xpub as a watch only wallet and Bluewallet shows a balance of 0 (even though this xpub has a balance) and it's because it imported it as BIP-44. Is there a way to tell Bluewallet which derivation path to use? Seems like this "checking for transactions to know which one to use" method is not working.

**Comment by ncoelho** [comment]
Advanced mode on > import > search accounts
